### DISCUSSION TITLE
creating FAST s-function: what should the directories be?

### DISCUSSION THREAD

**Discussion by liruiqi97** [OpenFAST/openfast#1034] [Q&A] [answered]
I am trying to operate the glue code for Simulink on MacBook. Before doing this, I have to create the s-function by running `create_FAST_SFunc.m` in folder `/Users/liruiqi/openfast/glue-codes/simulink/src`.

However, I am confused what should I set for the directories, e.g. the variables `installDir`, `libDir`, `includeDir`. The heading comment says that e.g. "the install directory is specified in the `installDir` variable", but i just have no idea what the install directory should be.

**Comment by bjonkman** [comment, marked as answer]
The `installDir` refers to the directory from the cmake `install` process where it copied the `openfastlib` library and associated files it needs to run. See section 2.2.4.2.2 of the [OpenFAST documentation](https://openfast.readthedocs.io/en/dev/source/install/index.html#compile-from-source) for how to set `CMAKE_INSTALL_PREFIX`.

Ideally, you wouldn't have to edit those directories in the create_FAST_SFunc.m file, but there have been some changes with Matlab on MacOS (and possibly the OpenFAST cmake defaults), so they may not be set to the correct defaults.

**Reply by liruiqi97** [reply]
thanks. let me explore the documentation
